### PR TITLE
refactor: reuse okhttp instance

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

as explained in [docs](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/) okHttp client must be shared across app sessions, to reduce memory footprint and improve latency 

### Solutions

we already share Http engine per user session so now we share it per app

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt [hunk 1]
```diff
@@ -37,67 +38,70 @@ import javax.net.ssl.SSLContext
...
-        val client = okHttpClient.build()
-        preconfigured = client
-        webSocketFactory = KaliumWebSocketFactory(client)
+    }.also {
+        preconfigured = it
+        webSocketFactory = KaliumWebSocketFactory(it)
     }
 }
 
-fun OkHttpClient.Builder.ignoreAllSSLErrors(): OkHttpClient.Builder {
```
Comment (MohamadJaara) [inline network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt:98]: this function was creating a new builder while it was not used

File: network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt [hunk 2]
```diff
@@ -37,67 +37,70 @@ import javax.net.ssl.SSLContext
...
+        if (ignoreSSLCertificates) ignoreAllSSLErrors()
 
-    if (certificatePinning.isNotEmpty()) {
-        val certPinner = CertificatePinner.Builder().apply {
-            certificatePinning.forEach { (cert, hosts) ->
-                hosts.forEach { host ->
-                    add(host, cert)
+        // OkHttp doesn't support configuring ping intervals dynamically,
+        // so they must be set when creating the Engine
+        // See https://youtrack.jetbrains.com/issue/KTOR-4752
```
Comment (typfel) [inline network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt:75]: Hasn't this comment lost its context? Shouldn't it be where the` pingInternval` is configured.

**Review by typfel — APPROVED** [review]
From what I can see it look alright.

**Review by Garzas — APPROVED** [review]

File: network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt (hunk 1 above)
Comment (yamilmedina) [reply network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt:98]: good catch !

**Review by yamilmedina — APPROVED** [review]
🍇 👌

**Pull request merged**